src/TRestAxionXrayWindow.cxx

```diff
@@ -0,0 +1,318 @@
+/// History of developments:
+///
+/// 2019-March: First concept and implementation of TRestAxionXrayWindow class.
+///             Javier Galan
+///
+/// \class      TRestAxionXrayWindow
+/// \author     Javier Galan
+///
+/// <hr>
+///
+
+#include "TRestAxionXrayWindow.h"
+using namespace std;
+
+#include "TRestSystemOfUnits.h"
+using namespace REST_Units;
+
+ClassImp(TRestAxionXrayWindow);
+
+///////////////////////////////////////////////
+/// \brief Default constructor
+///
+TRestAxionXrayWindow::TRestAxionXrayWindow() : TRestMetadata() { Initialize(); }
+
+///////////////////////////////////////////////
+/// \brief Constructor loading data from a config file
+///
+/// If no configuration path is defined using TRestMetadata::SetConfigFilePath
+/// the path to the config file must be specified using full path, absolute or
+/// relative.
+///
+/// The default behaviour is that the config file must be specified with
+/// full path, absolute or relative.
+///
+/// \param cfgFileName A const char* giving the path to an RML file.
+/// \param name The name of the specific metadata. It will be used to find the
+/// corresponding TRestGeant4Metadata section inside the RML.
+///
+TRestAxionXrayWindow::TRestAxionXrayWindow(const char* cfgFileName, string name)
+    : TRestMetadata(cfgFileName) {
+    debug << "Entering TRestAxionXrayWindow constructor( cfgFileName, name )" << endl;
+
+    Initialize();
+
+    LoadConfigFromFile(fConfigFileName, name);
+}
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestAxionXrayWindow::~TRestAxionXrayWindow() {}
+
+///////////////////////////////////////////////
+/// \brief Initialization of TRestAxionXrayWindow members. It removes all gases.
+///
+void TRestAxionXrayWindow::Initialize() {
+    SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
+    fEnergy.clear();
+    fTransmission.clear();
+}
+
+///////////////////////////////////////////////
+/// \brief It reads the data files from the corresponding material that
+/// needs to be found in the axiolib database with .sol extension.
+/// Usually placed under `data/axion/transmission/`
+///
+void TRestAxionXrayWindow::ReadMaterial() {
+    std::string materialFileName = SearchFile(fMaterial + ".sol");
+
+    debug << "TRestAxionXrayWindow::ReadMaterial. Reading file : " << materialFileName << endl;
+
+    if (!TRestTools::fileExists(materialFileName)) {
+        ferr << "TRestAxionXrayWindow::ReadMaterial( )" << endl;
+        ferr << "Material file not found : " << materialFileName << endl;
+        exit(1);
+    }
+
+    FILE* fin = fopen(materialFileName.c_str(), "rt");
+
+    fEnergy.clear();
+    fTransmission.clear();
+    double en, value;
+    while (fscanf(fin, "%lf\t%lf\n", &en, &value) != EOF) {
+        debug << "Energy : " << en << "eV -- Abs : " << value << endl;
+
+        fEnergy.push_back(en / 1000.);
+        fTransmission.push_back(TMath::Power(value, fThickness * units("um")));
```

Comment:
For what thickness are the stored data files? It's unclear from the file names. Just hard to reason whether the transmission calculation for a specific thickness makes sense that way.

Reply:
The transmission data (in the above code named `value`) is expressed in `um`. The fact that is expressed in `um` is described at the README.md in the data base.

https://github.com/rest-for-physics/axionlib-data/tree/master/transmission

I think it is important that the coding is correct here. But I believe this will be a rather static code that reads the file. If we add new materials those must be added for `1um`, and someone uploading new materials should be careful enough to read the README.md from the database.

So, basically, we get the transmission for 1um, for 2um you just need to square the transmission, right? Thats what the formula at line 189 is doing.

`fThickness` is expressed in `mm` because is the default units in REST, so thats why `units("um")` so that it is expressed finally in `us`.